src/Command/GenerateSitemapCommand.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SitemapPlugin\Command;
+
+use SitemapPlugin\Builder\SitemapBuilderInterface;
+use SitemapPlugin\Builder\SitemapIndexBuilderInterface;
+use SitemapPlugin\Filesystem\Writer;
+use SitemapPlugin\Renderer\SitemapRendererInterface;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+final class GenerateSitemapCommand extends Command
+{
+    /** @var \SitemapPlugin\Builder\SitemapBuilderInterface */
+    private $sitemapBuilder;
+
+    /** @var SitemapIndexBuilderInterface */
+    private $sitemapIndexBuilder;
+
+    /** @var SitemapRendererInterface */
+    private $sitemapRenderer;
+
+    /** @var SitemapRendererInterface */
+    private $sitemapIndexRenderer;
+
+    /** @var Writer */
+    private $writer;
+
+    /** @var ChannelRepositoryInterface */
+    private $channelRepository;
+
+    public function __construct(
+        SitemapRendererInterface $sitemapRenderer,
+        SitemapRendererInterface $sitemapIndexRenderer,
+        SitemapBuilderInterface $sitemapBuilder,
+        SitemapIndexBuilderInterface $sitemapIndexBuilder,
+        Writer $writer,
+        ChannelRepositoryInterface $channelRepository
+    ) {
+        $this->sitemapRenderer = $sitemapRenderer;
+        $this->sitemapIndexRenderer = $sitemapIndexRenderer;
+        $this->sitemapBuilder = $sitemapBuilder;
+        $this->sitemapIndexBuilder = $sitemapIndexBuilder;
+        $this->writer = $writer;
+        $this->channelRepository = $channelRepository;
+
+        parent::__construct('sylius:sitemap:generate');
+    }
+
+    protected function configure(): void
+    {
+        $this->addArgument('channel', InputArgument::IS_ARRAY, 'Channel codes to render. If none supplied, all channels will generated.');
```

Comment:
I was a bit in doubt here. Do you have some explanation on why it should for sure be an option?

Reply:
Okay, I did in fact believe that an arguments was required by default. I see that it's optional. That was why I wrote that. My personal preference for optional 'arguments' are `options` though. They don't have to come in any particular order and they are more explicit because you would have to write something like `--channel="CHANNEL"`

Reply:
Ok, sounds legit. Will adjust it to an option, as it adjusts behaviour instead of being required.